Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1

```diff
@@ -0,0 +1,344 @@
+
+    # The Original Owner and Repo in the AL-Go repository are microsoft/AL-Go-Actions, microsoft/AL-Go-PTE and microsoft/AL-Go-AppSource
+    $originalOwnerAndRepo = @{
+        "actionsRepo" = "microsoft/AL-Go-Actions"
+        "perTenantExtensionRepo" = "microsoft/AL-Go-PTE"
+        "appSourceAppRepo" = "microsoft/AL-Go-AppSource"
+    }
+    # Original branch is always main
+    $originalBranch = "main"
+
+    # Modify the file to use repository names based on whether or not we are using the direct AL-Go repo
+    $templateRepos = @{
+        "actionsRepo" = "AL-Go/Actions"
+        "perTenantExtensionRepo" = "AL-Go"
+        "appSourceAppRepo" = "AL-Go"
+    }
+
+    # Replace URL's to actions repository first
+    $regex = "^(.*)https:\/\/raw\.githubusercontent\.com\/microsoft\/AL-Go-Actions\/$originalBranch(.*)$"
+    $replace = "`${1}https://raw.githubusercontent.com/$($templateOwner)/AL-Go/$($templateBranch)/Actions`${2}"
+    $lines = $lines | ForEach-Object { $_ -replace $regex, $replace }
+
+    # Replace the owner and repo names in the workflow
+    "actionsRepo","perTenantExtensionRepo","appSourceAppRepo" | ForEach-Object {
+        $regex = "^(.*)$($originalOwnerAndRepo."$_")(.*)$originalBranch(.*)$"
+        $replace = "`${1}$($templateOwner)/$($templateRepos."$_")`${2}$($templateBranch)`${3}"
+        $lines = $lines | ForEach-Object { $_ -replace $regex, $replace }
+    }
+    $srcContent.Value = $lines -join "`n"
+}
+
+function IsDirectALGo {
+    param (
+        [string] $templateUrl
+    )
+    $directALGo = $templateUrl -like 'https://github.com/*/AL-Go@*'
+    if ($directALGo) {
+        if ($templateUrl -like 'https://github.com/microsoft/AL-Go@*' -and -not ($templateUrl -like 'https://github.com/microsoft/AL-Go@*/*')) {
+            throw "You cannot use microsoft/AL-Go as a template repository. Please use a fork of AL-Go instead."
+        }
+    }
+    return $directALGo
+}
+
+function GetSrcFolder {
+    Param(
+        [hashtable] $repoSettings,
+        [string] $templateUrl,
+        [string] $templateFolder,
+        [string] $srcPath
+    )
+    Write-Host $templateUrl
+    Write-Host $templateFolder
+    Write-Host $srcPath
+    if (!$templateUrl) {
+        return ''
+    }
+    if (IsDirectALGo -templateUrl $templateUrl) {
+        switch ($repoSettings.type) {
+            "PTE" {
+                $typePath = "Per Tenant Extension"
+            }
+            "AppSource App" {
+                $typePath = "AppSource App"
+            }
+            default {
+                throw "Unknown repository type"
+            }
+        }
+        $path = Join-Path $templateFolder "*/Templates/$typePath/$srcPath"
+    }
+    else {
+        $path = Join-Path $templateFolder "*/$srcPath"
+    }
+    Resolve-Path -Path $path -ErrorAction SilentlyContinue
+}
+
+function UpdateSettingsFile {
+    Param(
+        [string] $settingsFile,
+        [hashtable] $updateSettings,
+        [hashtable] $otherSettings = @{}

```

Comment:
Better name here could be `$addSettings` (or `$settingsToAdd`)

Reply:
Agree, changing to additionalSettings